lib/dpl/providers/scalingo.rb

```diff
@@ -8,30 +8,33 @@ class Scalingo < Provider
       required :api_key, [:username, :password]
 
       opt '--app APP', default: :repo_name
-      opt '--api_key KEY', 'scalingo API key', alias: :api_token, deprecated: :api_token
-      opt '--username NAME', 'scalingo username'
-      opt '--password PASS', 'scalingo password'
-      opt '--remote REMOTE', 'Git remote name', default: 'scalingo'
+      opt '--api_key KEY', 'Scalingo API key', alias: :api_token, deprecated: :api_token
+      opt '--username NAME', 'Scalingo username'
+      opt '--password PASS', 'Scalingo password'
+      opt '--region REGION', 'Scalingo region', default: 'agora-fr1', enum: %w(agora-fr1 osc-fr1)
+      opt '--remote REMOTE', 'Git remote name', default: 'scalingo-dpl'
       opt '--branch BRANCH', 'Git branch', default: 'master'
+      opt '--timeout SEC', 'Timeout for Scalingo CLI commands', default: 60
 
       needs :git, :ssh_key
 
-      cmds login_key:   'timeout 2 ./scalingo login --api-token %{api_key} > /dev/null 2>&1',
-           login_creds: 'echo -e \"%{username}\n%{password}\" | timeout 2 ./scalingo login > /dev/null 2>&1',
-           add_key:     './scalingo keys-add dpl_tmp_key %{key}',
-           remove_key:  './scalingo keys-remove dpl_tmp_key',
-           remote_add:  'git remote add %{remote} git@scalingo.com:%{app}.git > /dev/null 2>&1',
+      cmds login_key:   'timeout %{timeout} ./scalingo login --api-token %{api_key} > /dev/null 2>&1',
+           login_creds: 'echo -e \"%{username}\n%{password}\" | timeout %{timeout} ./scalingo login > /dev/null 2>&1',
+           add_key:     'timeout %{timeout} ./scalingo keys-add dpl_tmp_key %{key}',
+           remove_key:  'timeout %{timeout} ./scalingo keys-remove dpl_tmp_key',
+           git_setup:   './scalingo --app %{app} git-setup --remote %{remote}',
            push:        'git push %{remote} HEAD:%{branch} -f'
 
       errs install:    'Failed to install the Scalingo CLI.',
            login:      'Failed to authenticate with the Scalingo API.',
            add_key:    'Failed to add the ssh key.',
-           remote_add: 'Failed to add the git remote.',
-           push:       'Failed to push the app.',
-           remove_key: 'Failed to remove the ssh key.'
+           remove_key: 'Failed to remove the ssh key.',
+           git_setup:  'Failed to add the git remote.',
+           push:       'Failed to push the app.'
 
       def install
         script :install, assert: true
+        ENV['SCALINGO_REGION'] = region if region?
```

Comment:
```suggestion
        ENV['SCALINGO_REGION'] = region if region.empty?
```

Reply:
@EtienneM are you sure about this? We'd want to set the env var if `region` is present, right? The CLI args parser defines the method `region?` (which returns `true` if any region was given)

Reply:
I am going to merge this, as CI is green now after all. Please let me know if I misunderstood this here.